crates/typst/src/export/svg.rs

```diff
@@ -58,6 +62,73 @@ struct SVGRenderer {
     /// attribute of the group. The clip path is in the format of `M x y L x y C
     /// x1 y1 x2 y2 x y Z`.
     clip_paths: Deduplicator<EcoString>,
+    /// Deduplicated gradients with transform matrices. They use a reference
+    /// (`href`) instead of being defined inline. This saves a lot of space
+    /// since gradients are being reused but with different transforms can be
+    /// deduplicated.
+    gradient_refs: Deduplicator<GradientRef>,
+    /// These are the actual gradients being written in the SVG file.
+    /// These gradients are deduplicated because they do not contain the transform
+    /// matrix, allowing them to be reused across multiple invocations.
+    gradients: Deduplicator<(Gradient, Ratio)>,
```

Comment:
What is the `Ratio`?

Reply:
Added doc, basically it's the aspect ratio to correct the angle.